### ISSUE TITLE
Allow user to change location of OpenTTD data directory from ~/Documents

### ISSUE THREAD

**Issue by niklasva** [OpenTTD/OpenTTD#10125]
### Version of OpenTTD

OpenTTD 12.2, macOS 13.0

### Expected result

-

### Actual result

When the files in the data directory are updated, this will cause an automatic sync. Which happens a lot during gameplay.

I understand that this is more a problem with the nature of file syncing services, but it would be preferable if the data directory wasn't hard coded to be in the ~/Documents folder.
I would suggest changing it to ~/.openttd like on Linux, or even giving the user the ability to change it in settings.

### Steps to reproduce

Run the game while iCloud drive sync is turned on.

**Comment by glx22** [comment]
Probably autosave.
Yon can somehow control where autosaves ar stored, see https://github.com/OpenTTD/OpenTTD/blob/master/docs/directory_structure.md

**Comment by orudge** [comment]
Of course, folk who play OpenTTD on multiple Macs may well appreciate having their data synced.

**Comment by niklasva** [comment]
I have a suggestion on how we could solve this.

Many unix applications have a hierarchy of configuration files.
For instance, an application called foo would have a ~/.foorc file as its highest priority config file, then it would check ~/.foo/config, then ~/.config/foo and so on and so on.

I think making OpenTTD use a similar structure would be reasonable since the Linux port already has a ~/.openttd directory, and allowing the optional use of that same location on MacOS would make it more seamless to move between them.

TLDR; I suggest adding an optional data directory "~/.openttd" that would have a higher priority than "~/Documents/OpenTTD", but still use ~/Documents/OpenTTD as the default data folder.
